code_challenges/hacker_rank/exceptions.py

```diff
@@ -0,0 +1,25 @@
+def find_the_exception(number_of_test_cases, the_values):
+    result = []
+    for first, second in the_values:
+        try:
+            result.append(int(int(first) / int(second)))
+        except ZeroDivisionError:
+            result.append("Error Code: integer division or modulo by zero")
+        except ValueError:
+            if first.isdigit():
+                result.append(
+                    f"Error Code: invalid literal for int() with base 10: '{second}'"
+                )
+            else:
+                result.append(
+                    f"Error Code: invalid literal for int() with base 10: '{first}'"
+                )
+    return "\n".join(map(str, result))
```

Comment:
What is map doing for you here?  Is there a different approach that would work without using map?

Reply:
i want to print out a list on separate lines, and the items can sometimes be an int.
i cant use just join on a none string
what i did to change it was make the int a string after the division.
i could also do a loop but im not sure how to return all the results in a loop. if i have the return statement at the end of a for loop it will just exit after the first iteration